security/tpm/test/_tpm_test.js

```diff
@@ -116,13 +202,117 @@ describe('tpm', function () {
 
     it('correctly deals with error during tpm connection', function(done) {
       /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, the callback will be invoked with `err` of `InvalidOperationError`.] */
-      var client = new TpmSecurityClient('registration', fakeSimpleTpmClient);
+      var tpm = new tss.Tpm();
+      var client = new TpmSecurityClient('registration', tpm);
       client.signWithIdentity([1], (err, signResult) => {
         assert.isOk(err, 'should be an error from the connection portion of starting the tpm');
         done();
       });
     });
 
+    it('correctly deals with insufficient capabilities returned by the tpm', function(done) {
+      var fakeBadProps = {
+        tpmProperty: []
+      };
+      var tpm = new tss.Tpm();
+      var dataToSign = new Buffer('ab');
+      var connectStub = sinon.stub(tpm, 'connect');
+      connectStub.callsArgWith(0);
+      var client = new TpmSecurityClient(undefined, tpm);
+      var readPublicStub = sinon.stub(tpm, 'ReadPublic');
+      var lastResponseStub = sinon.stub(tpm, 'getLastResponseCode');
+      lastResponseStub.returns(tss.TPM_RC.SUCCESS);
+      var getPropsAndHashAlgStub = sinon.stub(client, '_getPropsAndHashAlg').callsArgWith(0,tss.TPM_ALG_ID.SHA256, fakeBadProps);
```

Comment:
>_getPropsAndHashAlg [](start = 55, length = 19)

still stubbing internals?

Reply:
This particular section of code will almost certainly change in some future rev.  Additionally, stubbing this provides basically no utility.

---
In reply to: [167386858](https://github.com/Azure/azure-iot-sdk-node/pull/225#discussion_r167386858) [](ancestors = 167386858)